src/System.Net.Ping/tests/FunctionalTests/UnixPingUtilityTests.cs

```diff
@@ -33,21 +34,46 @@ public class UnixPingUtilityTests
                 ? UnixCommandLinePing.Ping4UtilityPath
                 : UnixCommandLinePing.Ping6UtilityPath;
 
-            ProcessStartInfo psi = new ProcessStartInfo(utilityPath, arguments);
-            psi.RedirectStandardError = true;
-            psi.RedirectStandardOutput = true;
-            Process p = Process.Start(psi);
+            var p = new Process();
+            p.StartInfo.FileName = utilityPath;
+            p.StartInfo.Arguments = arguments;
+            p.StartInfo.UseShellExecute = false;
+            
+            p.StartInfo.RedirectStandardOutput = true;
+            var stdOutLines = new List<string>();
+            p.OutputDataReceived += new DataReceivedEventHandler(
+                delegate (object sendingProcess, DataReceivedEventArgs outputLine) { stdOutLines.Add(outputLine.Data); }); 
 
-            string pingOutput = p.StandardOutput.ReadToEnd();
-            Assert.True(p.WaitForExit(TestSettings.PingTimeout), "Ping process did not exit in " + TestSettings.PingTimeout + " ms.");
-            if (p.ExitCode == 1 || p.ExitCode == 2)
+            p.StartInfo.RedirectStandardError = true;
+            var stdErrLines = new List<string>();
+            p.ErrorDataReceived += new DataReceivedEventHandler(
+                delegate (object sendingProcess, DataReceivedEventArgs errorLine) { stdErrLines.Add(errorLine.Data); }); 
+
+            p.Start();
+            p.BeginOutputReadLine();
+            p.BeginErrorReadLine();
+
+            string pingOutput;
+            if (!p.WaitForExit(TestSettings.PingTimeout))
             {
-                // Workaround known OSX bug in ping6 utility.
-                Assert.Equal(utilityPath, UnixCommandLinePing.Ping6UtilityPath);
-                Assert.True(RuntimeInformation.IsOSPlatform(OSPlatform.OSX));
-                return;
+                pingOutput = string.Join("\n", stdOutLines);
+                string stdErr = string.Join("\n", stdErrLines);
+                throw new Exception(
+                    $"[{utilityPath} {arguments}] process did not exit in {TestSettings.PingTimeout} ms.\nStdOut:[{pingOutput}]\nStdErr:[{stdErr}]");
             }
 
+            // Ensure standard output and error are flushed
+            p.WaitForExit();
+
+            pingOutput = string.Join("\n", stdOutLines);
+            var exitCode = p.ExitCode;
+            if (exitCode != 0)
```

Comment:
There can be stderr output with a zero exit code.

Reply:
Good point, but if it matters to us parse will fail and then it is added to the exception.